src/event/log_event.rs

```diff
@@ -78,6 +80,41 @@ impl LogEvent {
     pub fn is_empty(&self) -> bool {
         self.fields.is_empty()
     }
+
+    #[instrument(skip(self, lookup), fields(lookup = %lookup), err)]
+    fn entry(&mut self, lookup: Lookup) -> crate::Result<Entry<String, Value>> {
+        trace!("Seeking to entry.");
+        let mut walker = lookup.into_iter().enumerate();
+
+        let mut current_pointer = if let Some((index, Segment::Field(segment))) = walker.next() {
+            trace!(%segment, index, "Seeking segment.");
+            self.fields.entry(segment)
+        } else {
+            return Err("Lookups should have at least one segment.".into());
+        };
+
+        for (index, segment) in walker {
+            trace!(%segment, index, "Seeking next segment.");
+            current_pointer = match (segment, current_pointer) {
+                (Segment::Field(field), Entry::Occupied(entry)) => match entry.into_mut() {
+                    Value::Map(map) => map.entry(field),
+                    v => return Err(format!("Looking up field on a non-map value: {:?}", v).into()),
+                },
+                (Segment::Field(field), Entry::Vacant(entry)) => {
+                    trace!(segment = %field, index, "Met vacant entry.");
+                    return Err(format!(
+                        "Tried to step into `{}` of `{}`, but it did not exist.",
+                        field,
+                        entry.key()
+                    )
```

Comment:
What would cause this to happen exactly? As a user of the API I'm not exactly sure how I would handle errors like this.

Reply:
Say I said `foo.bar.baz.bat` but only `foo.bar` existed, this would tell you "Tried to step into `baz` of `bar` but it did not exist", so you could diagnose.